content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,673 @@
+
+If you do need to make a change, you should plan to either:
+- change the name of the ConfigMap, and switch to running Pods that reference the new name
+- replace all the nodes in your cluster that have previously run a Pod that used the old value
+- restart the kubelet on any node where the kubelet previously loaded the old ConfigMap
+{{< /note >}}
+
+An example manifest for an [Immutable ConfigMap](/docs/concepts/configuration/configmap/#configmap-immutable) is shown below.
+{{% code_sample file="configmap/immutable-configmap.yaml" %}}
+
+Create the Immutable ConfigMap:
+```shell
+kubectl apply -f https://k8s.io/examples/configmap/immutable-configmap.yaml
+```
+
+Below is an example of a Deployment manifest with the Immutable ConfigMap `company-name-20150801` mounted as a
+{{< glossary_tooltip text="volume" term_id="volume" >}} into the Pod's only container.
+{{% code_sample file="deployments/deployment-with-immutable-configmap-as-volume.yaml" %}}
+
+Create the Deployment:
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-immutable-configmap-as-volume.yaml
+```
+
+Check the pods for this Deployment to ensure they are ready (matching by
+{{< glossary_tooltip text="selector" term_id="selector" >}}):
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=immutable-configmap-volume
+```
+
+You should see an output similar to:
+```
+NAME                                          READY   STATUS    RESTARTS   AGE
+immutable-configmap-volume-78b6fbff95-5gsfh   1/1     Running   0          62s
+immutable-configmap-volume-78b6fbff95-7vcj4   1/1     Running   0          62s
+immutable-configmap-volume-78b6fbff95-vdslm   1/1     Running   0          62s
+```
+
+The Pod's container refers to the data defined in the ConfigMap and uses it to print a report to stdout. You can check this report by viewing the logs for one of the Pods in that Deployment:
+```shell
+# Pick one Pod that belongs to the Deployment, and view its logs
+kubectl logs deployments/immutable-configmap-volume
+```
+
+You should see an output similar to:
+```
+Found 3 pods, using pod/immutable-configmap-volume-78b6fbff95-5gsfh
+Wed Mar 20 03:52:34 UTC 2024 The name of the company is ACME, Inc.
+Wed Mar 20 03:52:44 UTC 2024 The name of the company is ACME, Inc.
+Wed Mar 20 03:52:54 UTC 2024 The name of the company is ACME, Inc.
+```
+
+{{< note >}}
+Once a ConfigMap is marked as immutable, it is not possible to revert this change nor to mutate the contents of the data or the binaryData field.  
+In order to modify the contents, we will change the name of the ConfigMap, and switch to running Pods that reference the new name.
+{{< /note >}}
+
+Edit the ConfigMap:
+```shell
+kubectl edit configmap company-name-20150801
+```
+
+In the editor that appears, perform the following actions:
+- change the name of the ConfigMap from `company-name-20150801` to `company-name-20240312`
```

Comment:
does this really work?

Reply:
All instructions have been tested by me and they work successfully.